### DISCUSSION TITLE
HREF and JS links not working in compiled production build - OK in Dev

### DISCUSSION THREAD

**Discussion by Blatman** [wailsapp/wails#1773] [Q&A] [answered]
OS: 10.13 and 12.5(amd64 on M1 Pro)

Trying to encapsulate a legacy web using <script type="text/javascript" src="js/linktest.js"></script> pattern.
Using wails dev works fine - the vite server uses localhost:34115 - images, page links and js all work OK in the Dev window and with normal browser pointing at 34115.
Using wails build the app builds OK and when run the pages display OK with all images however the page links don't work and the JS is non-functional.
In dev console (debug build) I notice that resources are not found eg: wails://wails/js/linktest.js and wails://wails/page1.html

The vanilla WAILS template works OK with production builds but that uses the ES module pattern.

Is it the case that production built apps will only support HTML/JS code using the ES module pattern but Dev will support the older style?

**Comment by leaanthony** [comment]
Yes - all assets should be part of the `dist` directory if you want your app to find them in a production build. In your case, you could use something like this: https://www.npmjs.com/package/vite-plugin-copy-files

**Comment by Blatman** [comment, marked as answer]
Well - back from Tauri as I want to get Wails going so I can see which will be best to use from a frontend JS perspective - anyway I  tried the above plugin but couldn't get a result however I did find an answer on StackOverflow.

https://stackoverflow.com/questions/69540063/vite-static-files-are-not-copying

The important snippet -
You can place the asset in a special public directory under your project root. Assets in this directory will be served at root path / during dev, and copied to the root of the dist directory as-is.

I placed the app files in a public folder under src (which apparently is the default Project Root) and the build then worked OK with the files being transferred to the dist folder by the build process.
Note: I still have the vite copy plugin installed but I don't enact it since the .vite.config.js I had to create for testing has been removed.

It looks like the in place Vite server uses the default src and dist folders ( in fact the Wails example files are in the src folder).
This means that legacy web apps that use classic linking of assets should now be able to be packaged with Wails quite easily.

So - onward with the testing - the milestones upto 2.5.0 look appealing.

**Reply by leaanthony** [reply]
That's great and welcome back! Would love some insights into how we could improve. Do you think you would like to update the documentation with this information? Maybe a guide? No worries if not

**Reply by Blatman** [reply]
Probably not at this stage since I am still trying to understand how it all works so I am trying to build a basic vanilla JS app to demonstrate the various features without having to work out all sorts of exotic frameworks (Vue/React and all the rest) just to access the basic Wails stuff.
One niggle with docs I find (not only with this repo) is that it is not always obvious whether particular example snippets of code are JS or Go so I need to get them into complete files and see where they fit in and get them working.
